test/test_torch.py

```diff
@@ -7644,33 +7643,28 @@ def _test_where_scalar_template(self, device, dtype, exec_fn):
         for with_extremal in [True, False]:
             for ndims in range(0, 4):
                 shape = self._rand_shape(ndims, min_size=5, max_size=10)
-                for n in range(ndims + 1):
-                    for c in combinations(list(range(ndims)), n):
-                        for scalar_type in [int, float, complex]:
-                            if dtype.is_complex:
-                                condition = self._generate_input(shape, dtype, device, with_extremal).abs() > 0.5
-                            else:
-                                condition = self._generate_input(shape, dtype, device, with_extremal) > 0.5
-
-                            x = self._generate_input(shape, dtype, device, with_extremal)
-
-                            if not dtype.is_complex and scalar_type == complex:
-                                continue
+                for scalar_type in [int, float, complex]:
+                    if dtype.is_complex:
+                        # Use make_tensor when it has a flag to add extremals.
+                        condition = self._generate_input(shape, dtype, device, with_extremal).abs() > 0.5
+                    else:
+                        # Use make_tensor when it has a flag to add extremals.
+                        condition = self._generate_input(shape, dtype, device, with_extremal) > 0.5
 
-                            scalar_1 = scalar_type(random.random())
+                    # Use make_tensor when it has a flag to add extremals.
+                    x = self._generate_input(shape, dtype, device, with_extremal)
+                    scalar_1 = scalar_type(random.random())
 
-                            exec_fn(scalar_type, dtype, condition, x, scalar_1)
+                    exec_fn(scalar_type, dtype, condition, x, scalar_1)
 
     # For current implementation,
     # below are the valid `TensorDtype` and `ScalarType` combinations.
     def _where_valid_scalar_tensor_combination(self, scalar_type, dtype):
-        if (scalar_type == int and dtype == torch.long):
+        if (dtype.is_floating_point):
+            return scalar_type in (int, float)
+        elif dtype.is_complex:
```

Comment:
Super tiny nit: if/if would be better here (no elif) since the if path returns

Reply:
Done :)